File: cpp/Binary Tree/L_rightSideView.cpp
#include<bits/stdc++.h>
using namespace std;

 struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 };

vector<int> traverse(TreeNode* node, vector<int> res, int depth) {
    if(node==NULL) return res;
    if(res.size()==depth) res.push_back(node->val);
    res = traverse(node->right, res, depth + 1);
    res = traverse(node->left,  res, depth + 1);
    return res;
}

vector<int> rightSideView(TreeNode* root) {
    vector<int> res;
    res = traverse(root, res, 0);
    return res;
}

int main() {

}

File: Python/Binary Trees/L_rightSideView.py
def rightSideView(root):
        res = []
        def traverse(node, depth):
            if not node:
                return None
            if len(res) <= depth:
                res.append(node.val)
            traverse(node.right, depth + 1)
            traverse(node.left,  depth + 1)
        
        traverse(root, 0)
        return res


def rightSideView(root):
        res = []
        def dfs(root, curr):
            if not root: return
            if curr == len(res):
                res.append(root.val)
            dfs(root.right, curr + 1)
            dfs(root.left, curr + 1)
        dfs(root, 0)
        return res